EpsilonAuxBmsV2/Core/Src/Tasks/OrionInterfaceTask.c

```diff
@@ -0,0 +1,178 @@
+#include "OrionInterfaceTask.h"
+
+void orionInterfaceTask(void* arg)
+{
+    OrionCanInfo message = (OrionCanInfo)
+    {
+        0
+    };
+
+    for (;;)
+    {
+        orionInterface(&message);
+    }
+}
+
+/*
+Essentially reads inputs from Orion and gets the Aux BMS to respond accordingly.
+Determines:
+ * When car should trip (all contactors open)
+ * When charge/discharge should be open or closed based on GPIO inputs coming from Orion and based on Max/Min cell voltages
+
+It will get info from the CAN Rx Interrupt parser periodically, and each time it does it will set the orionCanMessageReceivedRecently member of AuxStatus to 1 and keep track of the last time it received a message
+If the waiting on the queue times out, that means that no message from Orion has been received recently, so set the orionCanMessageReceivedRecently member of AuxStatus to 0
+
+Before it finishes or triggers events to control the contactors, it updates the AuxTrip variable, and the AuxStatus variable.
+To update the AuxStatus variable, it must first acquire the auxStatusOrionInterfaceMutex
+To update the AuxTrip variable, it must first acquire the auxTripMutex
+
+It will check the ContactorState variable before it requests a contactor to open or close (no point closing if already closed)
+The reason the gatekeeper tasks don’t have to be periodic is because all additional control will come from here.
+If a contactor is open, but it is allowed to be closed, the startupDone member of ContactorState indicates the startup sequence has finished, and Orion was the reason the contactor is open, Orion Interface will request that the contactor is closed.
+Will only ever be charge or discharge. If common is open after startup, the car would have tripped.
+When opening a contactor, it temporarily raises the gatekeeper task’s priority to RealTime (highest priority)
+*/
+void orionInterface(OrionCanInfo* message)
+{
+    osStatus_t status = osMessageQueueGet(orionInterfaceQueue, message, NULL, ORION_QUEUE_TIMEOUT);
+
+
+    uint8_t shouldDisconnectContactors = 0;
+    // Read Orion charge and discharge enable GPIO
+    uint8_t orionDischargeEnableSense = HAL_GPIO_ReadPin(ORION_DISCHARGE_ENABLE_SENSE_GPIO_Port, ORION_DISCHARGE_ENABLE_SENSE_Pin);
+    uint8_t orionChargeEnableSense = HAL_GPIO_ReadPin(ORION_CHARGE_ENABLE_SENSE_GPIO_Port, ORION_CHARGE_ENABLE_SENSE_Pin);
+    // Set aux status and aux trip to default values
+    localAuxStatus = (AuxStatus)
```

Comment:
Because of unit testing. We're passing the addresses of those variables into the trip functions, and since those are Mocked, we need the address argument to match what we provide in the `os_Expect_And_Return`